Utilities/build_ubuntu_cross_compilation_toolchain

```diff
@@ -244,17 +244,19 @@ fi
 # fix up glibc modulemap
 fix_glibc_modulemap "$cross_tc_basename/$xc_tc_name/usr/lib/swift/linux/x86_64/glibc.modulemap"
 
-cat > "$cross_tc_basename/ubuntu-xenial-destination.json" <<EOF
+cat > "$cross_tc_basename/ubuntu-jammy-destination.json" <<EOF
 {
     "version": 1,
     "sdk": "$(pwd)/$cross_tc_basename/$linux_sdk_name",
     "toolchain-bin-dir": "$(pwd)/$cross_tc_basename/$xc_tc_name/usr/bin",
-    "target": "x86_64-unknown-linux",
+    "target": "x86_64-unknown-linux-gnu",
     "extra-cc-flags": [
         "-fPIC"
     ],
     "extra-swiftc-flags": [
-        "-use-ld=lld", "-tools-directory", "$(pwd)/$cross_tc_basename/$xc_tc_name/usr/bin"
+        "-use-ld=lld",
+        "-tools-directory", "$(pwd)/$cross_tc_basename/$xc_tc_name/usr/bin",
+        "-sdk", "$(pwd)/$cross_tc_basename/$linux_sdk_name",
```

Comment:
@neonichu interesting that this flag wasn't previously needed. And I'd assume it always has the same value as the `sdk` property on the top level of `destination.json`, so maybe we could propagate it to `swiftc` in SwiftPM internally? Maybe it was propagated before, but somehow regressed or something else changed?

Reply:
I didn't even know the `-sdk` parameter was a thing on Linux, tbh. What issues are you seeing when not passing `-sdk`?

Reply:
AFAIU the sysroot path is inferred from `-sdk`. Without it can't link `crt` files:

```
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: cannot open Scrt1.o: No such file or directory
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: cannot open crti.o: No such file or directory
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: cannot open crtbeginS.o: No such file or directory
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: unable to find library -lgcc
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: unable to find library -lgcc_s
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: unable to find library -lc
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: unable to find library -lgcc
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: unable to find library -lgcc_s
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: cannot open crtendS.o: No such file or directory
/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld: error: cannot open crtn.o: No such file or directory
```

Here's the final linker invocation without `-sdk`:

```
"/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld" 
  --sysroot=/Applications/Xcode-14.0.0-Release.Candidate.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk 
  -pie --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2
  -o ~/macro-test/.build/x86_64-unknown-linux-gnu/debug/macro-test
  Scrt1.o crti.o crtbeginS.o -L/tmp/cross-toolchain/swift.xctoolchain/usr/lib/swift/linux
  -L/Applications/Xcode-14.0.0-Release.Candidate.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib/swift -L~/macro-test/.build/x86_64-unknown-linux-gnu/debug
  -L/tmp/cross-toolchain/swift.xctoolchain/usr/lib -L/Applications/Xcode-14.0.0-Release.Candidate.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib
  -z nostart-stop-gc -rpath /tmp/cross-toolchain/swift.xctoolchain/usr/lib/swift/linux
  -rpath /Applications/Xcode-14.0.0-Release.Candidate.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib/swift
  /tmp/cross-toolchain/swift.xctoolchain/usr/lib/swift/linux/x86_64/swiftrt.o
  ~/macro-test/.build/x86_64-unknown-linux-gnu/debug/macro_test.build/macro_test.swift.o
  ~/macro-test/.build/x86_64-unknown-linux-gnu/debug/macro_test.build/macro_test.swiftmodule.o
  -lswift_Concurrency -lswiftCore -lswiftSwiftOnoneSupport -lswiftCore
  --defsym main=macro_test_main "-rpath=\$ORIGIN" -lgcc --as-needed -lgcc_s --no-as-needed
  -lc -lgcc --as-needed -lgcc_s --no-as-needed crtendS.o crtn.o
```

Notice how references to `MacOSX12.3.sdk` are present everywhere, but there are no prefixes for `crt` files whatsoever. I guess `MacOSX12.3.sdk` is inferred as the default SDK on macOS when `-sdk` option is not passed?

For comparison, here's an invocation when correct cross-compilation `-sdk` is passed:

```
"/tmp/cross-toolchain/swift.xctoolchain/usr/bin/ld.lld"
  --sysroot=/tmp/cross-toolchain/ubuntu-jammy.sdk
  -pie --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2
  -o ~/macro-test/.build/x86_64-unknown-linux-gnu/debug/macro-test 
  /tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/x86_64-linux-gnu/Scrt1.o 
  /tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/x86_64-linux-gnu/crti.o 
  /tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/gcc/x86_64-linux-gnu/12/crtbeginS.o
  -L/tmp/cross-toolchain/swift.xctoolchain/usr/lib/swift/linux
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/swift
  -L~/macro-test/.build/x86_64-unknown-linux-gnu/debug
  -L/tmp/cross-toolchain/swift.xctoolchain/usr/lib 
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/gcc/x86_64-linux-gnu/12 
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/lib/x86_64-linux-gnu 
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/lib/../lib64
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/x86_64-linux-gnu
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/lib
  -L/tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib
  -z nostart-stop-gc 
  -rpath /tmp/cross-toolchain/swift.xctoolchain/usr/lib/swift/linux 
  -rpath /tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/swift 
  /tmp/cross-toolchain/swift.xctoolchain/usr/lib/swift/linux/x86_64/swiftrt.o 
  ~/macro-test/.build/x86_64-unknown-linux-gnu/debug/macro_test.build/macro_test.swift.o 
  ~/macro-test/.build/x86_64-unknown-linux-gnu/debug/macro_test.build/macro_test.swiftmodule.o 
  -lswift_Concurrency -lswiftCore -lswiftSwiftOnoneSupport -lswiftCore
  --defsym main=macro_test_main "-rpath=\$ORIGIN" -lgcc --as-needed -lgcc_s --no-as-needed
  -lc -lgcc --as-needed -lgcc_s --no-as-needed
  /tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/gcc/x86_64-linux-gnu/12/crtendS.o 
  /tmp/cross-toolchain/ubuntu-jammy.sdk/usr/lib/x86_64-linux-gnu/crtn.o
```

Reply:
Ah OK, I think that suggests that we should probably just pass `-sdk` automatically.

Reply:
I hope you don't mind then if I submit that as a separate PR?

Reply:
That sounds fine, I think we can land this one as-is and remove the `-sdk` option from here when we made the underlying change.

Reply:
This appears to be a mac-specific bug, as I don't need the redundant flag when [cross-compiling from linux using the Android NDK sysroot](https://github.com/buttaface/swift-android-sdk/blob/main/android-aarch64.json).